### PR TITLE
Use tag codes on the server (filing for jimmyfnugent)

### PR DESCRIPTION
**Pull request by hopskipnfall**
- client now expects tag codes (chars, stages, labels) from the server
- client now uses full Stages etc for everything in state
- filters mostly work
- getting Bits from the server is broken until we migrate the ones in mongo to use the codes

### CODE REVIEW

File: client/src/uri_util.ts [hunk 1]
```diff
@@ -101,7 +102,7 @@ export const buildUriFromState = (state: AppState) => {
     [QUERY_SORT]: defaultToUndefined(state.filtering.sort),
     [QUERY_LIMIT]: defaultToUndefined(state.filtering.limit),
     [QUERY_OFFSET]: defaultToUndefined(state.filtering.offset),
-    [QUERY_MAIN_CHARS]: defaultToUndefined(getCharFilterQuery(state.filtering.mainCharacters)),
+    [QUERY_MAIN_CHARS]: defaultToUndefined(getCharFilterQuery(new Set(Array.from(state.filtering.mainCharacters).map((char: Character) => char.id)))),
```
Comment (hopskipnfall) [inline client/src/uri_util.ts:105]: `(char: Character)` should just be `char`. It should infer the type.

File: client/src/thunks.ts [hunk 2]
```diff
@@ -6,16 +7,16 @@ import { AppState } from "./store";
...
 import { buildUriFromState } from "./uri_util";
 
 type AppThunkAction = ThunkAction<Promise<void>, AppState, null, AnyAction>
 type AppThunkActionCreator = ActionCreator<AppThunkAction>
 
 export const thunkFetchBits: AppThunkActionCreator = () => {
   return async (dispatch, getState) => {
-    const response = await fetchBitsApi(getState().filtering)
-    dispatch(replaceBits(response.bits));
+    const bits: Bit[] = await fetchBitsApi(getState().filtering);
```
Comment (hopskipnfall) [inline client/src/thunks.ts:18]: This should be inferred. Consider adding a return type to fetchBitsApi if it's ambiguous.

File: client/src/thunks.ts [hunk 3]
```diff
@@ -1,3 +1,4 @@
+import * as _ from "lodash";
```
Comment (hopskipnfall) [inline client/src/thunks.ts:1]: i think using lodash throws types away and replaces with `any`. if that's the case i'm not really a fan of using this library.

File: client/src/uri_util.ts (hunk 1 above)
Comment (jimmyfnugent) [reply client/src/uri_util.ts:105]: Done.

File: client/src/thunks.ts (hunk 2 above)
Comment (jimmyfnugent) [reply client/src/thunks.ts:18]: Done.

File: client/src/thunks.ts (hunk 3 above)
Comment (jimmyfnugent) [reply client/src/thunks.ts:1]: I tried changing one of the types and it complained (both VSCode and the TS compiler), so I think we're okay.

**Pull request merged**